packages/graphqlsp/src/ast/index.ts

```diff
@@ -54,45 +59,13 @@ function resolveIdentifierToGraphQLCall(
   info: ts.server.PluginCreateInfo,
   checker: ts.TypeChecker | undefined
 ): checks.GraphQLCallNode | null {
-  let prevElement: ts.Node | undefined;
-  let element: ts.Node | undefined = input;
-  // NOTE: Under certain circumstances, resolving an identifier can loop
-  while (ts.isIdentifier(element) && element !== prevElement) {
-    prevElement = element;
+  if (!checker) return null;
 
-    const definitions = info.languageService.getDefinitionAtPosition(
-      element.getSourceFile().fileName,
-      element.getStart()
-    );
+  const value = getValueOfIdentifier(input, checker);
+  if (!value) return null;
```

Comment:
nit: redundant `null` conversions, which can be avoided if we adjust signatures here